Modules/Applications/AppDomainTransform/app/otbDomainTransform.cxx

```diff
@@ -0,0 +1,453 @@
+    int mode = GetParameterInt("mode");
+
+    if( dir != 0 && dir != 1)
+      {
+      itkExceptionMacro(<< "-dir is '"
+         << dir << "'."
+         << "It must be either 'fwd' or 'inv'");
+      }
+
+    if( mode != 0 && mode != 1)
+      {
+      itkExceptionMacro( << "mode is '"
+         << mode << "'."
+         << "It must must be either 'fft' or 'wavelet'");
+      }
+
+    if ( mode == 1)
+      {
+      int wavelet_type = GetParameterInt("mode.wavelet.form");
+      unsigned int nlevels = GetParameterInt("mode.wavelet.nlevels");
+      switch (wavelet_type)
+        {
+        case 0:
+          {
+          DoWaveletTransform<otb::Wavelet::HAAR>(dir, nlevels);
+          break;
+          }
+        case 1:
+          {
+          DoWaveletTransform<otb::Wavelet::DB4>(dir, nlevels);
+          break;
+          }
+        case 2:
+          {
+          DoWaveletTransform<otb::Wavelet::DB4>(dir, nlevels);
+          break;
+          }
+        case 3:
+          {
+          DoWaveletTransform<otb::Wavelet::DB6>(dir, nlevels);
+          break;
+          }
+        case 4:
+          {
+          DoWaveletTransform<otb::Wavelet::DB8>(dir, nlevels);
+          break;
+          }
+        case 5:
+          {
+          DoWaveletTransform<otb::Wavelet::DB12>(dir, nlevels);
+          break;
+          }
+        case 6:
+          {
+          DoWaveletTransform<otb::Wavelet::DB20>(dir, nlevels);
+          break;
+          }
+        case 7:
+          {
+          DoWaveletTransform<otb::Wavelet::SPLINE_BIORTHOGONAL_2_4 >(dir, nlevels);
+          break;
+          }
+        case 8:
+          {
+          DoWaveletTransform<otb::Wavelet::SPLINE_BIORTHOGONAL_4_4>(dir, nlevels);
+          break;
+          }
+        case 9:
+          {
+          DoWaveletTransform<otb::Wavelet::SYMLET8>(dir, nlevels);
+          break;
+          }
+        default:
+          {
+          itkExceptionMacro( << "Invalid wavelet type: '" <<  wavelet_type << "'");
+          break;
+          }
+        }
+      }
+    else
+      {
+      // fft ttransform
+      bool shift = IsParameterEnabled( "mode.fft.shift");
+
+      if (dir == 0 )
+        {
+        //forward fft
+        typedef otb::Image<TInputPixel>          TInputImage;
+        typedef TInputImage::Pointer TInputImagePointer;
+
+        //get input paramter as otb::Image<TInputPixel>
+        TInputImagePointer inImage = GetParameterImage<TInputImage>("in");
+        inImage->UpdateOutputInformation();
+        //typedef itk::::ForwardFFTImageFilter over otbImage< TInputPixel >
+        typedef itk::ForwardFFTImageFilter < TInputImage> FFTFilter;
+        FFTFilter::Pointer fwdFilter = FFTFilter::New();
+        fwdFilter->SetInput( inImage );
+
+        //typedef VectorImage for output of UnaryFunctorImageFilter
+        typedef otb::VectorImage<TOutputPixel>          TOutputImage;
+
+        //UnaryFunctorImageFilter for Complex to VectorImage
```

Comment:
After searching, there is already a filter to do the conversion, try using otb::ComplexToVectorImageCastFilter (in Module/Core/Common)